### PR TITLE
Access result rows as namedtuples

### PR DESCRIPTION
**Pull request by amCap1712**
Builds on #2119

This PR fixes the warning:

Using non-integer/slice indices on Row is deprecated and will be removed in version 2.0; please use row._mapping[<key>], or the mappings() accessor on the Result object.
(Background on SQLAlchemy 2.0 at: https://sqlalche.me/e/b8d9)

For rationale, see section: https://docs.sqlalchemy.org/en/14/changelog/migration_20.html#result-rows-act-like-named-tuples

### CODE REVIEW

File: listenbrainz/db/dump.py [hunk 1]
```diff
@@ -671,7 +671,7 @@ def get_dump_entries():
               ORDER BY created DESC
             """))
 
-        return [dict(row) for row in result]
```
Comment (alastair) [inline listenbrainz/db/dump.py:674]: try `result.mappings().all()` or `[dict(row) for row in result.mappings()]`

File: listenbrainz/db/dump.py [hunk 2]
```diff
@@ -684,7 +684,7 @@ def get_dump_entry(dump_id):
             'dump_id': dump_id,
         })
         if result.rowcount > 0:
-            return dict(result.fetchone())
+            return result.fetchone()._asdict()
```
Comment (alastair) [inline listenbrainz/db/dump.py:687]: try `return result.mappings().first()`

**Review by alastair — COMMENTED** [review]
We had a discussion in IRC about this, and mentioned that it'd be nicer to use the `result.mappings()` method to return us dictionary-looking items when we need dictionaries, so that we don't have to use `_hidden` style variables. I think that in the cases where we just get a few values and return them we can use this new style (e.g. `return Session(row.id, row.user_id, row.sid, row.api_key, row.ts)` is OK by me)

**Comment by amCap1712** [comment]
Just to clarify, `_asdict()` is named like a private member but isn't one. Its intended for public use but has been named with an _ for "reasons".

File: listenbrainz/db/dump.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz/db/dump.py]: Done

File: listenbrainz/db/dump.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/db/dump.py:710]: Done

**Pull request merged**